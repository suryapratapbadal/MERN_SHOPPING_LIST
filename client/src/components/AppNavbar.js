import React, { Component } from 'react';
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    Container,
    Button,
} from 'reactstrap';

export default class AppNavbar extends Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen : false,
        };

    }

    toggle = () => {
        this.setState({
            isOpen: !this.state.isOpen
        });
    }

    render() {
        return (
            <div>
                <Navbar color="dark" dark expand="sm" className="mb-5">
                    <Container>
                        <NavbarBrand href="/">ShoopinList</NavbarBrand>
                        <NavbarToggler onClick={this.toggle} />
                        <Collapse isOpen={this.state.isOpen} navbar>
                            <Nav className="ml-auto" navbar>
                                <NavItem>
                                    <NavLink href="https://github.com/suryapratapbadal/MERN_SHOPPING_LIST">Github</NavLink>
                                </NavItem>
                            </Nav>
                            <Button color='secondary' style={{marginLeft: '2em'}} disabled={!this.props.user}>Log Out</Button>
                        </Collapse>
                    </Container>
                </Navbar>
            </div>
        );
    }
}