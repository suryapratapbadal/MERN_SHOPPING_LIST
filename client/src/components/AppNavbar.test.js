import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import AppNavbar from './AppNavbar';

// Test to check , Component Renders
describe('<AppNavbar />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<AppNavbar />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});


// Test to check toggle function onClick

describe('<AppNavbar />', () => {
  describe('onClick()', () => {
    test('successfully calls the onClick handler', () => {
      const mockOnClick = jest.fn();
      const wrapper = shallow(
        <AppNavbar />
      );


      const toggleState = wrapper.find('Collapse').props().isOpen;
      wrapper.find('NavbarToggler').simulate('click');

      expect(wrapper.find('Collapse').props().isOpen).toEqual(!toggleState);
    });
  });
});