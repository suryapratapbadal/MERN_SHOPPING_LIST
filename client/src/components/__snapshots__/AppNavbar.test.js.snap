// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppNavbar /> render() renders the component 1`] = `
<div>
  <Navbar
    className="mb-5"
    color="dark"
    dark={true}
    expand="sm"
    tag="nav"
  >
    <Container
      tag="div"
    >
      <NavbarBrand
        href="/"
        tag="a"
      >
        ShoopinList
      </NavbarBrand>
      <NavbarToggler
        onClick={[Function]}
        tag="button"
        type="button"
      />
      <Collapse
        appear={false}
        enter={true}
        exit={true}
        in={false}
        isOpen={false}
        mountOnEnter={false}
        navbar={true}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        tag="div"
        timeout={350}
        unmountOnExit={false}
      >
        <Nav
          className="ml-auto"
          navbar={true}
          tag="ul"
          vertical={false}
        >
          <NavItem
            tag="li"
          >
            <NavLink
              href="https://github.com/suryapratapbadal/MERN_SHOPPING_LIST"
              tag="a"
            >
              Github
            </NavLink>
          </NavItem>
        </Nav>
      </Collapse>
    </Container>
  </Navbar>
</div>
`;
